runtime-decompiler/src/main/java/org/jrd/frontend/frame/settings/SettingsView.java

```diff
@@ -55,43 +69,150 @@
 
             chooser = new JFileChooser();
             File dir;
-            if (isPortable()) {
-                dir = new File(getJrdLocation() + File.separator + "libs");
+            if (Directories.isPortable()) {
+                dir = new File(Directories.getJrdLocation() + File.separator + "libs");
             } else {
-                dir = new File(getJrdLocation() + File.separator + "decompiler_agent" + File.separator + "target");
+                dir = new File(Directories.getJrdLocation() + File.separator +
+                        "decompiler_agent" + File.separator + "target");
             }
-            chooser.setCurrentDirectory(fallback(dir));
+            chooser.setCurrentDirectory(FileSelectorArrayRow.fallback(dir));
 
             this.setLayout(new GridBagLayout());
             GridBagConstraints gbc = new GridBagConstraints();
             gbc.anchor = GridBagConstraints.WEST;
             gbc.fill = GridBagConstraints.BOTH;
 
-            gbc.insets = new Insets(0,20,0,0);
+            addAgent(gbc);
+
+            // Nested Jars
+            nestedJars = new JLabel("Nested Jars Settings:");
+            newExtensionsTextField = new JTextField();
+            newExtensionsTextField.addKeyListener(new KeyListener() {
+                @Override
+                public void keyTyped(KeyEvent keyEvent) {
+
+                }
+
+                @Override
+                public void keyPressed(KeyEvent keyEvent) {
+                    if (keyEvent.getKeyCode() == KeyEvent.VK_ENTER) {
+                        confirmExtensions();
+                    }
+                }
+
+                @Override
+                public void keyReleased(KeyEvent keyEvent) {
+
+                }
+            });
+
+            defaultListModel = new DefaultListModel<>();
+            currentExtensionsList = new JList<>(defaultListModel);
+            currentExtensionsList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+            currentExtensionsList.setLayoutOrientation(JList.VERTICAL);
+            currentExtensionsList.setVisibleRowCount(-1);
+
+            scrollPane = new JScrollPane(currentExtensionsList);
+            scrollPane.setPreferredSize(new Dimension(0, 200));
+
+            addButton = new JButton("Add");
+            addButton.addActionListener(actionEvent -> confirmExtensions());
+
+            removeButton = new JButton("Remove");
+            removeButton.addActionListener(actionEvent -> defaultListModel.removeElementAt(currentExtensionsList.getSelectedIndex()));
```

Comment:
When nothing is selected, this prints an ugly trace. Please catch the exception and log it appropriately.